bbb/services.py

```diff
@@ -379,6 +356,31 @@ def __init__(self, pulse_queue_basename, pulse_exchange_basename, worker_type,
         )
         super(TCListener, self).__init__(*args, events=events, **kwargs)
 
+    def _isAuthorized(self, buildername, scopes):
+        """Tests to see if the builder given is restricted, and if so, whether
+        or not the scopes given are authorized to use it. Builders that do
+        not match the overall restricted builder patterns do not require any
+        scopes. Builders that do must have a buildbot-bridge:builder-name:
+        scope that matches the builder name given."""
+        for r in self.restricted_builders:
+            if re.match(r, buildername):
+                break
+        # If not restricted
+        else:
+            return True
+
+        # If it's a restricted builder, we need to check scopes!
+        for s in scopes:
+            if not s.startswith("buildbot-bridge:builder-name:"):
+                continue
+            allowed_builder = s.split(":")[-1]
+            if buildername == allowed_builder or allowed_builder == "*":
+                return True
+            elif allowed_builder.endswith("*") and buildername.startswith(allowed_builder[:-1]):
+                return True
```

Comment:
That is a _much_ nicer implementation indeed, thank you! It passed the unit tests I wrote for my own implementation, too. I think I'll just wait until that PR is merged though...no point in landing it in this repo too :)